### PR TITLE
Upgrade Armeria version to 0.98.0

### PR DESCRIPTION
**Pull request by ikhoon**
Modifications:
* Migrate all deprecated or removed API in Armeria 0.98.0
* Change `WatchTimeout.makeReasonable` to `WatchTimeout.availableTimeout` 
  to use new timeout methods introduced in Armeria 0.98.0
* Make `ChangesRequestConverter` and `CommitMessageRequestConverter` 
  implement `RequestConverterFunction`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,6 +14,9 @@ allprojects {
     repositories {
         mavenCentral()
         jcenter()
+        //maven {
+        //    url 'https://oss.sonatype.org/content/repositories/snapshots/'
+        //}
```
Comment (trustin) [inline build.gradle]: Intentionally left commented so a dev can easily try snapshot builds.

**Comment by ikhoon** [comment]
Let me resolve the conflict

**Comment by ikhoon** [comment]
But... @trustin did already 🤣

**Comment by trustin** [comment]
But we have a new conflict here. Thanks in advance :-)

**Comment by ikhoon** [comment]
Sure! I will take a look.

**Pull request merged**